### ISSUE TITLE
CurrencyType Not Supported When Rollups Triggered by Community User

### ISSUE THREAD

**Issue by jesslopez-sf** [SalesforceFoundation/Cumulus#3298]
When a Community User creates an Opportunity and our rollups run, a System.QueryException error is returned because the Community User does not have access to the CurrencyType object. "System.QueryException: sObject type 'CurrencyType' is not supported."

**Trace:**
npsp.TDTM_Opportunity: execution of AfterUpdate
caused by: System.QueryException: sObject type 'CurrencyType' is not supported.
Class.npsp.UTIL_CurrencyCache.getOrgDefaultCurrency: line 142, column 1
Class.npsp.UTIL_Currency.getOrgDefaultCurrency: line 107, column 1
Class.npsp.RLLP_OppRollup_UTIL: line 72, column 1
Class.npsp.RLLP_OppRollup.<init>: line 42, column 1
Class.npsp.RLLP_OppRollup_TDTM.run: line 60, column 1
Class.npsp.TDTM_TriggerHandler.runClass: line 183, column 1
Class.npsp.TDTM_TriggerHandler.run: line 105, column 1
Trigger.npsp.TDTM_Opportunity: line 34, column 1

The error occurs in UTIL_CurrencyCache when we attempt to query the CurrencyType object:
```
String soql = 'SELECT IsoCode FROM CurrencyType WHERE IsCorporate = True LIMIT 1';
SObject currObj = database.query(soql); 
```

A potential solution would be to run in a "without sharing" context.

**Comment by jesslopez-sf** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x8000000AYAwBAAX" target="_blank">W-025550</a>

**Comment by mrbelvedere** [comment]
Included in beta release 3.131 (Beta 11)

**Comment by mrbelvedere** [comment]
Included in production release 3.131
